ext/src/CoupledSDEs.jl

```diff
@@ -0,0 +1,206 @@
+        p,
+        noise_strength;
+        diffeq = diffeq,
+        noise_rate_prototype = noise_rate_prototype,
+        noise = noise,
+        seed = seed
+    )
+end
+
+"""
+    CoupledODEs(ds::CoupledSDEs; kwargs...)
+
+Converts a [`CoupledSDEs`](@ref) into [`CoupledODEs`](@ref).
+"""
+function DynamicalSystemsBase.CoupledODEs(
+        sys::CoupledSDEs; diffeq = DEFAULT_DIFFEQ, t0 = 0.0)
+    return CoupledODEs(
+        dynamic_rule(sys), SVector{length(sys.integ.u)}(sys.integ.u), sys.p0;
+        diffeq = diffeq, t0 = t0
+    )
+end
+
+# Pretty print
+function DynamicalSystemsBase.additional_details(ds::CoupledSDEs)
+    solver, remaining = _decompose_into_sde_solver_and_remaining(ds.diffeq)
+    return [
+        "Noise strength" => ds.noise_strength,
+        "SDE solver" => string(nameof(typeof(solver))),
+        "SDE kwargs" => remaining,
+        "Noise type" => ds.noise_type
+    ]
+end
+
+###########################################################################################
+# API - obtaining information from the system
+###########################################################################################
+
+SciMLBase.isinplace(::CoupledSDEs{IIP}) where {IIP} = IIP
+StateSpaceSets.dimension(::CoupledSDEs{IIP, D}) where {IIP, D} = D
+DynamicalSystemsBase.current_state(ds::CoupledSDEs) = current_state(ds.integ)
+DynamicalSystemsBase.isdeterministic(ds::CoupledSDEs) = false
+
+function DynamicalSystemsBase.dynamic_rule(ds::CoupledSDEs)
+    # with remake it can happen that we have nested SDEFunctions
+    # sciml integrator deals with this internally well
+    f = ds.integ.f
+    while hasfield(typeof(f), :f)
+        f = f.f
+    end
+    return f
+end
+
+# so that `ds` is printed
+function DynamicalSystemsBase.set_state!(ds::CoupledSDEs, u::AbstractArray)
+    (set_state!(ds.integ, u); ds)
+end
+SciMLBase.step!(ds::CoupledSDEs, args...) = (step!(ds.integ, args...); ds)
+
+# TODO We have to check if for SDEIntegrators a different step interruption is possible.
+function DynamicalSystemsBase.successful_step(integ::AbstractSDEIntegrator)
+    rcode = integ.sol.retcode
+    return rcode == SciMLBase.ReturnCode.Default || rcode == SciMLBase.ReturnCode.Success
+end
+
+# This is here to ensure that `u_modified!` is called
+function DynamicalSystemsBase.set_parameter!(ds::CoupledSDEs, args...)
+    _set_parameter!(ds, args...)
+    u_modified!(ds.integ, true)
+    return nothing
+end
+
+DynamicalSystemsBase.referrenced_sciml_prob(ds::CoupledSDEs) = ds.integ.sol.prob
+
+###########################################################################################
+# Utilities
+###########################################################################################
+
+function σg(σ, g)
+    return (u, p, t) -> σ .* g(u, p, t)
+end
+
+function σg!(σ, g!)
+    function (du, u, p, t)
+        g!(du, u, p, t)
+        du .*= σ
+        return nothing
+    end
+end
+
+function add_noise_strength(σ, g, IIP)
```

Comment:
I'd rename this to `multiply_noise_strength()` or `scale_noise_term()`

Reply:
not used in new interface